cloud/amazon/ec2_ami.py

```diff
@@ -72,9 +72,10 @@
     default: null
   delete_snapshot:
     description:
-      - Whether or not to delete an AMI while deregistering it.
+      - Whether or not to delete snapshots while deregistering AMI. 
     required: false
-    default: null
+    default: "no"
+    choices: [ "yes", "no" ] 
   tags:
     description:
       - a hash/dictionary of tags to add to the new image; '{"key":"value"}' and '{"key":"value","key":"value"}'
```

Comment:
- A dictionary of tags to add to the new image; '{"key":"value"}' and '{"key":"value","key":"value"}'